Compilation error on Ubuntu 18.04

On Ubuntu 18.04 I got this error: 
~~~
[ 96%] Building CXX object apps/cymoca/CMakeFiles/cymoca.dir/cymoca.cpp.o
/home/straversaro/src/cymoca/apps/cymoca/cymoca.cpp: In function ‘int main(int, const char**)’:
/home/straversaro/src/cymoca/apps/cymoca/cymoca.cpp:95:5: error: reference to ‘ifstream’ is ambiguous
     ifstream fileStream(model_path.string());
     ^~~~~~~~
In file included from /usr/include/boost/filesystem/string_file.hpp:14:0,
                 from /usr/include/boost/filesystem.hpp:19,
                 from /home/straversaro/src/cymoca/apps/cymoca/cymoca.cpp:2:
/usr/include/boost/filesystem/fstream.hpp:169:32: note: candidates are: typedef class boost::filesystem::basic_ifstream<char> boost::filesystem::ifstream
   typedef basic_ifstream<char> ifstream;
                                ^~~~~~~~
In file included from /usr/include/c++/7/ios:38:0,
                 from /usr/include/c++/7/ostream:38,
                 from /usr/include/c++/7/iostream:39,
                 from /home/straversaro/src/cymoca/apps/cymoca/cymoca.cpp:1:
/usr/include/c++/7/iosfwd:162:34: note:                 typedef class std::basic_ifstream<char> std::ifstream
   typedef basic_ifstream<char>   ifstream;
                                  ^~~~~~~~
/home/straversaro/src/cymoca/apps/cymoca/cymoca.cpp:96:24: error: ‘fileStream’ was not declared in this scope
     cymoca::Compiler c(fileStream);
                        ^~~~~~~~~~
apps/cymoca/CMakeFiles/cymoca.dir/build.make:62: recipe for target 'apps/cymoca/CMakeFiles/cymoca.dir/cymoca.cpp.o' failed
make[2]: *** [apps/cymoca/CMakeFiles/cymoca.dir/cymoca.cpp.o] Error 1
CMakeFiles/Makefile2:331: recipe for target 'apps/cymoca/CMakeFiles/cymoca.dir/all' failed
make[1]: *** [apps/cymoca/CMakeFiles/cymoca.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
~~~
I guess something changed in the recent version of boost shipped with Ubuntu 18.04 w.r.t. to the one shipped in Travis. 

This commit: https://github.com/traversaro/cymoca/commit/b6642c616ec057effe63688839aba54c21b83372 seems to fix it.

--- a/apps/cymoca/cymoca.cpp
+++ b/apps/cymoca/cymoca.cpp
@@ -5,7 +5,7 @@
 #include "cymoca_compiler/version.h"
 #include "cymoca_compiler/listener/LispPrinter.h"
 
-using namespace boost::filesystem;
+namespace fs = boost::filesystem;
 namespace po = boost::program_options;
 using namespace std;
 
@@ -35,8 +35,6 @@ int main(int argc, const char *argv[]) {
 
     /** Define and parse the program options 
      */
-    namespace po = boost::program_options;
-
     po::options_description generic("Generic options");
     generic.add_options()
         ("help,h", "print help message");
@@ -90,7 +88,7 @@ int main(int argc, const char *argv[]) {
     }
 
     // compile
-    path model_path(vm["model"].as<string>());
+    fs::path model_path(vm["model"].as<string>());
     assert(exists(model_path));
     ifstream fileStream(model_path.string());
     cymoca::Compiler c(fileStream);

--- a/scripts/format.sh
+++ b/scripts/format.sh
@@ -18,7 +18,7 @@ then
 else
 	clang-format-6.0 -output-replacements-xml $files | grep -c "<replacement " >/dev/null
 	if [ $? -ne 1 ]; then 
-    	echo "Commit did not match clang-format"
+    	echo "Commit did not match clang-format, please run '$0 1'"
 		git clang-format-6.0 --diff HEAD^ $files
     	exit 1
 	fi

--- a/scripts/format.sh
+++ b/scripts/format.sh
@@ -19,7 +19,7 @@ else
 	clang-format-6.0 -output-replacements-xml $files | grep -c "<replacement " >/dev/null
 	if [ $? -ne 1 ]; then 
     	echo "Commit did not match clang-format"
-		git-clang-format-6.0 --diff
+		git clang-format-6.0 --diff HEAD^ $files
     	exit 1
 	fi
 fi

--- a/test/test_casadi.cpp
+++ b/test/test_casadi.cpp
@@ -7,9 +7,9 @@
 #include "cymoca_compiler/listener/LispPrinter.h"
 #include "cymoca_compiler/listener/WhenExpander.h"
 
-using namespace boost::filesystem;
 using namespace cymoca;
 using namespace std;
+namespace fs = boost::filesystem;
 namespace ca = casadi;
 
 class CasadiListener : public ast::ConstListener {
@@ -168,7 +168,7 @@ TEST(CasadiTest, BasicFunction) {
 }
 
 TEST(CasadiTest, BouncingBall) {
-  path p("../../test/models/BouncingBall.mo");
+  fs::path p("../../test/models/BouncingBall.mo");
   ASSERT_TRUE(exists(p));
   ifstream fileStream(p.string());
   cymoca::Compiler c(fileStream);

--- a/test/test_parse.cpp
+++ b/test/test_parse.cpp
@@ -7,7 +7,7 @@
 
 using namespace std;
 using namespace cymoca;
-namespace fs=boost::filesystem;
+namespace fs = boost::filesystem;
 
 TEST(ParseTest, BouncingBall) {
   cout << endl;

--- a/test/test_parse.cpp
+++ b/test/test_parse.cpp
@@ -5,22 +5,22 @@
 #include <cymoca_compiler/Compiler.h>
 #include <cymoca_compiler/listener/LispPrinter.h>
 
-using namespace boost::filesystem;
+using namespace std;
 using namespace cymoca;
+namespace fs=boost::filesystem;
 
 TEST(ParseTest, BouncingBall) {
-  std::cout << std::endl;
-  path p("../../test/models/BouncingBall.mo");
+  cout << endl;
+  fs::path p("../../test/models/BouncingBall.mo");
   ASSERT_TRUE(exists(p));
-  std::ifstream fileStream(p.string());
+  ifstream fileStream(p.string());
   Compiler c(fileStream);
-  // std::cout << s<< std::endl;
   EXPECT_STREQ("Hello World!", "Hello World!");
 
   listener::LispPrinter listener;
   Walker walker;
   walker.walk(*c.root(), listener);
-  std::cout << listener.get() << std::endl;
+  cout << listener.get() << endl;
 }
 
 // vim: set et fenc=utf-8 ff=unix sts=0 sw=2 ts=2 :

